File: 99_bottles.py
def bottle_message(bottle_count):
    result = ""
    
    if bottle_count > 1:
        result = str(bottle_count) + " bottles"
    elif bottle_count == 1:
        result = str(bottle_count) + " bottle"
    else:
        result = "no more bottles"
    
    return result

def bottle_line_one(bottle_count):
    MESSAGE = bottle_message(bottle_count)
    return f"{MESSAGE} of beer on the wall, {MESSAGE} of beer.".capitalize();

def bottle_line_two(bottle_count):
    MESSAGE = bottle_message(bottle_count)
    return f"Take one down and pass it around, {MESSAGE} of beer on the wall.\n"

def bottle_line_final(bottle_count):
    MESSAGE = bottle_message(bottle_count)
    return "Go to the store and buy some more, " + MESSAGE + " of beer on the wall."

def bottleSong():
    TOTAL_BOTTLES = 99
  
    bottle_count = TOTAL_BOTTLES
    while bottle_count  >= 1:
        print(bottle_line_one(bottle_count))
        print(bottle_line_two(bottle_count - 1))
        bottle_count -= 1
    
    print(bottle_line_one(bottle_count))
    print(bottle_line_final(TOTAL_BOTTLES))

bottleSong()

File: 99_bottles.js
bottleSong();

function bottleSong() 
{
  const TOTAL_BOTTLES = 20;
  
  let bottleCount = TOTAL_BOTTLES;
  for(; bottleCount >= 1; --bottleCount)
  {
      console.log(bottleLineOne(bottleCount));
      console.log(bottleLineTwo(bottleCount - 1));
  }
  
  console.log(bottleLineOne(bottleCount));
  console.log(bottleLineFinal(TOTAL_BOTTLES));
}

function bottleLineOne(bottleCount)
{
    const message = bottleMessage(bottleCount);
    return capitalize(`${message} of beers on the wall, ${message} of beer`);
}

function bottleLineTwo(bottleCount)
{
    const message = bottleMessage(bottleCount);
    return `Take one down and pass it around, ${message} of beer on the wall.\n`
}

function bottleMessage(bottleCount)
{
    let result = "";
    
    if(bottleCount > 1)
    {
        result = bottleCount + " bottles";
    }
    else if (bottleCount === 1)
    {
        result = bottleCount + " bottle";
    }
    else
    {
        result = "no more bottles";
    }
    
    return result;
}

function capitalize(string)
{
    return string[0].toUpperCase() + string.substring(1).toLowerCase();
}

function bottleLineFinal(bottleCount)
{
    const message = bottleMessage(bottleCount);
    return "Go to the store and buy some more, " + message + " of beer on the wall."
}